### PR TITLE
[Arith] Inverse affine map

### PR DESCRIPTION
**Pull request by vinx13**
This PR added a utility function to perform inverse transformation for the bijective affine map. This will be helpful for tensorization in the future.

@tqchen

### CODE REVIEW

**Review by tqchen — APPROVED** [review]

File: tests/python/unittest/test_arith_iter_affine_map.py [hunk 1]
```diff
@@ -643,6 +643,67 @@ def test_normalize_iter_map_to_expr():
...
+
+    # simple case
+    l0_0, l0_1 = isplit(l0, 16)
+    l1_0, l1_1 = isplit(l1, 4)
+    l0_1_l1_1_fused = ifuse([l0_1, l1_1])
+
+    iter_map = tvm.arith.detect_iter_map([l0_1_l1_1_fused[0], l0_0[0], l1_0[0]], var_dom([l0, l1]))
+    outputs = [tvm.tir.Var("output_{}".format(i), "int32") for i in range(len(iter_map))]
+    res = tvm.arith.inverse_affine_iter_map(iter_map, outputs)
+    print(res)
```
Comment (spectrometerHBH) [inline tests/python/unittest/test_arith_iter_affine_map.py:660]: this line can be removed

File: src/arith/iter_affine_map.cc [hunk 2]
```diff
@@ -1385,5 +1385,147 @@ TVM_REGISTER_GLOBAL("arith.SubspaceDivide")
...
+        if (!base_scale.defined() || op->value < base_scale->value) {
+          base_scale = GetRef<IntImm>(op);
+          base_index = i;
+        }
+      }
+    }
+    ICHECK(base_scale.defined());
+    std::vector<IterSplitExpr> iters;
+    std::vector<bool> visited(sum_expr->args.size(), false);
+    PrimExpr expected_scale = base_scale;
```
Comment (spectrometerHBH) [inline src/arith/iter_affine_map.cc:1502]: I think we can expect the sum expression has its children in descending order, so to search for base scale is not necessary

Comment (vinx13) [reply src/arith/iter_affine_map.cc:1502]: if the sum expression is produced by `DetectIterMap` this is true, can we can simply this whole function, I can add some assertions to check this assumption @tqchen

**Review by spectrometerHBH — APPROVED** [review]

File: src/arith/iter_affine_map.cc (hunk 2 above)
Comment (masahi) [reply src/arith/iter_affine_map.cc:1502]: sorry I've just merged it...

**Comment by tqchen** [comment]
@vinx13 let us followup with another PR to address the last comment @spectrometerHBH bought up

**Pull request merged**